pvlib/tests/test_spectrum.py

```diff
@@ -123,24 +123,70 @@ def test_get_example_spectral_response():
     assert_allclose(sr, expected, rtol=1e-5)
 
 
+@fail_on_pvlib_version('0.12')
 def test_get_am15g():
     # test that the reference spectrum is read and interpolated correctly
     e = spectrum.get_am15g()
     assert_equal(len(e), 2002)
     assert_equal(np.sum(e.index), 2761442)
     assert_approx_equal(np.sum(e), 1002.88, significant=6)
 
-    wavelength = [270, 850, 950, 1200, 4001]
-    expected = [0.0, 0.893720, 0.147260, 0.448250, 0.0]
+    wavelength = [270, 850, 950, 1200, 1201.25, 4001]
+    expected = [0.0, 0.893720, 0.147260, 0.448250, 0.4371025, 0.0]
 
     e = spectrum.get_am15g(wavelength)
     assert_equal(len(e), len(wavelength))
     assert_allclose(e, expected, rtol=1e-6)
 
 
+@pytest.mark.parametrize(
+    "reference_identifier,expected_sums",
+    [
+        (
+            "ASTM G173-03",  # reference_identifier
+            {  # expected_sums
+                "extraterrestrial": 1356.15,
+                "global": 1002.88,
+                "direct": 887.65,
+            },
+        ),
+    ],
+)
+def test_get_standard_spectrum(reference_identifier, expected_sums):
+    # test reading of reference spectrum
+    standard = spectrum.get_standard_spectrum(standard=reference_identifier)
+    assert set(standard.columns) == expected_sums.keys()
+    assert standard.index.name == "wavelength"
+    assert standard.index.is_monotonic_increasing is True
+    expected_sums = pd.Series(expected_sums)  # convert prior to comparison
+    assert_series_equal(np.sum(standard), expected_sums, atol=1e-2)
+
+
+def test_get_standard_spectrum_custom_wavelengths():
+    # test that the spectrum is interpolated correctly when custom wavelengths
+    # are specified
+    # only checked for ASTM G173-03 reference spectrum
+    wavelength = [270, 850, 951.634, 1200, 4001]
+    expected_sums = pd.Series(
+        {"extraterrestrial": 2.23266, "global": 1.68952, "direct": 1.58480}
+    )  # for given ``wavelength``
+    standard = spectrum.get_standard_spectrum(
+        wavelength, standard="ASTM G173-03"
+    )
+    assert_equal(len(standard), len(wavelength))
+    assert any(standard.isna())  # check no NaN values were returned
```

Comment:
```suggestion
    assert not any(standard.isna())  # check no NaN values were returned
```

Should it be this?

Reply:
More or less, it seems the any evaluation is made on the iterator of a pandas dataframe, which is the columns names, not the values. Well spotted the incoherent test code.
https://stackoverflow.com/questions/65255622/anydf-isna-returns-true-when-no-nans-are-present